src/main/java/org/broadinstitute/hellbender/tools/spark/bwa/BwaSpark.java

```diff
@@ -40,15 +47,25 @@ public boolean requiresReads() {
     }
 
     @Override
-    protected void runTool(final JavaSparkContext ctx) {
+    public List<ReadFilter> getDefaultReadFilters() {
+        // 1) unmapped or neither secondary nor supplementary and 2) has some sequence
+        return Collections.singletonList(ReadFilterLibrary.PRIMARY_ALIGNMENT.and(ReadFilterLibrary.SEQ_IS_STORED));
```

Comment:
We might also want to include `ReadFilterLibrary.HAS_MATCHING_BASES_AND_QUALS` since we can detect that here and we would screen those reads out downstream.